tests/unit/test-extension-runners/test.chromium.js

```diff
@@ -357,4 +368,71 @@ describe('util/extension-runners/chromium', async () => {
     await runnerInstance.exit();
   });
 
+  describe('reloadAllExtensions', () => {
+    let runnerInstance;
+    let wsClient;
+
+    beforeEach(async () => {
+      const {params} = prepareExtensionRunnerParams();
+      runnerInstance = new ChromiumExtensionRunner(params);
+      await runnerInstance.run();
+    });
+
+    const connectClient = async () => {
+      const wssInfo = runnerInstance.wss.address();
+      const wsURL = `ws://${wssInfo.address}:${wssInfo.port}`;
+      wsClient = new WebSocket(wsURL);
+      await new Promise((resolve) => wsClient.on('open', resolve));
+    };
+
+    afterEach(async () => {
+      wsClient.close();
+      await runnerInstance.exit();
+    });
+
+    it('does not resolve before complete message from client', async (done) => {
+      let reloadMessage = false;
+      await connectClient();
+      const waitCompleteMessage = new Promise((resolve) => {
+        wsClient.on('message', (message) => {
+          const msg = JSON.parse(message);
+
+          if (msg.type === 'webExtReloadExtensionComplete') {
+            reloadMessage = true;
+            resolve();
+          }
+        });
+      });
+      setTimeout(async () => {
+        assert.equal(reloadMessage, false);
+        const msg = JSON.stringify({type: 'webExtReloadExtensionComplete'});
+        wsClient.send(msg);
+        await waitCompleteMessage;
+        assert.equal(reloadMessage, true);
+        done();
+      }, 350);
+      await runnerInstance.reloadAllExtensions();
+    });
+
+    it('resolve when any client send complete message', async () => {
+      await connectClient();
+      wsClient.on('message', () => {
+        const msg = JSON.stringify({type: 'webExtReloadExtensionComplete'});
+        wsClient.send(msg);
+      });
+      assert.isOk(await runnerInstance.reloadAllExtensions());
+    });
+
+    it('resolve when all client disconnect', async () => {
+      await connectClient();
+      wsClient.on('close', async () => {
+        assert.isOk(await runnerInstance.reloadAllExtensions());
```

Comment:
This implementation is incorrect. The test will finish as soon as `wsClient.close();` has been called. You should call `reloadAllExtensions()`, save its return value in a variable, call `wsClient.close()` and then wait for the resolution of the promise that you've saved in the variable.